parser/src/main/java/org/dst/parser/DstNewSqlListener.java

```diff
@@ -20,6 +21,27 @@ public DstParsedResult getParsedResult() {
     return parsedResult;
   }
 
+  @Override
+  public void enterStrPut(DstNewSQLParser.StrPutContext ctx) {
+    Preconditions.checkState(parsedResult == null);
+    Preconditions.checkState(ctx.children.size() == 3);
+
+    StringProtocol.PutRequest.Builder builder = StringProtocol.PutRequest.newBuilder();
+    builder.setKey(ctx.children.get(1).getText());
+    builder.setValue(ctx.children.get(2).getText());
+    parsedResult = new DstParsedResult(RequestTypeEnum.STR_PUT, builder.build());
+  }
+
+  @Override
+  public void enterStrGet(DstNewSQLParser.StrGetContext ctx) {
+    Preconditions.checkState(parsedResult == null);
+    Preconditions.checkState(ctx.children.size() == 2);
```

Comment:
it's necessary to check childern size beacuse garammer will be checked,if error it will not access this method,i feel it's redundent

Reply:
Thanks. I think it's important to assert the children size because this layer is not able to ensure the size is correct.